pandas/tests/scalar/test_timestamp.py

```diff
@@ -1079,6 +1079,25 @@ def test_is_leap_year(self):
             dt = Timestamp('2100-01-01 00:00:00', tz=tz)
             assert not dt.is_leap_year
 
+    def test_timestamp(self):
+        # GH#17329
+        # tz-naive --> treat it as if it were UTC for purposes of timestamp()
+        ts = Timestamp.now()
+        uts = ts.replace(tzinfo=utc)
+        assert ts.timestamp() == uts.timestamp()
+
+        tsc = Timestamp('2014-10-11 11:00:01.12345678', tz='US/Central')
+        utsc = tsc.tz_convert('UTC')
+        # utsc is a different representation of the same time
+        assert tsc.timestamp() == utsc.timestamp()
+
+        if PY3:
+            # should agree with datetime.timestamp method
+            dt = ts.to_pydatetime()
+            assert dt.timestamp() == round(ts.timestamp(), 6)
```

Comment:
why do you need to round?

Reply:
Because `dt.timestamp()` is the `datetime.timestamp` method which only has microsecond-precision.

Reply:
exactly, you should *not* need to do this